app/models/division.rb

```diff
@@ -124,9 +125,14 @@ def parent_division_and_name
   end
 
   def generate_short_name
-    return if short_name.present? && Division.pluck(:short_name).exclude?(self.short_name)
+    # no change to short_name that is already saved and therefore uniq
+    return if self.short_name.present? && self.short_name == self.attribute_in_database(:short_name)
```

Comment:
Could you use `persisted? && !short_name_changed?` Anything around attrib changes should probably used the AR Dirty features.